docs/dub-reference/dependencies.md

```diff
@@ -0,0 +1,25 @@
+# Dependency syntax
+
+## version
+
+See [version specifiers](./build_settings#version-specifiers).
+
+### `"*"`
+### `~>a.b.c`
+### `~>a.b`
+### `>=a`, `>=a <b`, `<b`
```

Comment:
I think I would recommend changing the titles to `version="..."` for all of them

alternatively this might just be better as a plain list with some short descriptions, since we are likely not going to write much about each of them. (writing less is better for things that the user can infer from previous documentation and just experience)

I think also adding a table showing what matches here would make it easier to understand as well, such as:

<table>
<tr>
 <td>matches?
 <td> (no release)
 <td> v0.0.1
 <td> v0.1.0
 <td> v0.2.0
 <td> v1.0.0
 <td> v2.0.0
<tr>
 <td><code>version="*"</code>
 <td> ✔️
 <td> ✔️
 <td> ✔️
 <td> ✔️
 <td> ✔️
 <td> ✔️
<tr>
 <td><code>version="~>0.1.0"</code>
 <td>
 <td>
 <td> ✔️
 <td>
 <td>
 <td>
<tr>
 <td><code>version="~>0.1"</code>
 <td>
 <td>
 <td> ✔️
 <td> ✔️
 <td>
 <td>
<tr>
 <td><code>version=">=0.1.0"</code>
 <td>
 <td>
 <td> ✔️
 <td> ✔️
 <td> ✔️
 <td> ✔️
<tr>
 <td><code>version=">=0.1.0 <1.0.0"</code>
 <td>
 <td>
 <td> ✔️
 <td> ✔️
 <td>
 <td>
<tr>
 <td><code>version="<2.0.0"</code>
 <td>
 <td> ✔️
 <td> ✔️
 <td> ✔️
 <td> ✔️
 <td>
</table>

Reply:
I added the table, though it might be better to put that in `build_settings` with the info about how the version matching works. Then we can just link from `dependencies` rather than repeating anything.

Reply:
this verbose extra info is definitely something that should be in dependencies rather than in build settings, which should only give an overview about available keys with concise documentation. I think the list of what each syntax means is enough.